src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReplIdeFeatures.cs

```diff
@@ -24,14 +24,19 @@ public CSharpReplIdeFeatures(VisualStudioInstanceFactory instanceFactory)
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
+            try
             {
-                VisualStudio.Workspace.SetUseSuggestionMode(false);
-                VisualStudio.InteractiveWindow.ClearReplText();
-                VisualStudio.InteractiveWindow.Reset();
+                if (disposing && VisualStudio != null)
```

Comment:
Should we have a separate method to override (Cleanup() or the like) that does this pattern automatically?

Reply:
😆 So, it turns out @333fred called this ages ago (https://github.com/dotnet/roslyn/pull/25467#discussion_r174545764). `DisposeAsync` is only called if `InitializeAsync` completes successfully, so these methods should be paired for resource management, and the constructor initialization should be paired with `Dispose`.

Reply:
For some reason a bunch of tests failed when I made the switch in this PR, so I'm going to put it in the follow-up PR instead.